### DISCUSSION TITLE
json::parse string is order

### DISCUSSION THREAD

**Discussion by ApacheWang** [nlohmann/json#2234] [Q&A] [answered]
Hello nlohmann, I have a question, when using json::parse string, the output json string is sorted by key, is there a way to let json save the key sequence of the original string

**Comment by gregmarr** [comment]
Answered in the FAQ https://nlohmann.github.io/json/home/faq/

**Reply by nlohmann** [reply]
To quote:

> By default, the library does not preserve the **insertion order of object elements**. This is standards-compliant, as the [JSON standard](https://tools.ietf.org/html/rfc8259.html) defines objects as "an unordered collection of zero or more name/value pairs".
> 
> If you do want to preserve the insertion order, you can specialize the object type with containers like [`tsl::ordered_map`](https://github.com/Tessil/ordered-map) ([integration](https://github.com/nlohmann/json/issues/546#issuecomment-304447518)) or [`nlohmann::fifo_map`](https://github.com/nlohmann/fifo_map) ([integration](https://github.com/nlohmann/json/issues/485#issuecomment-333652309)).

**Reply by nlohmann** [reply]
@ApacheWang Do you need further assistance?

**Comment by nlohmann** [comment, marked as answer]
When #2258 is merged, it will be possible to use type `ordered_json` which maintains the insertion order of object keys.

**Reply by nlohmann** [reply]
Done.

**Comment by gbartl** [comment]
I am trying to do the same thing here and I am getting loads of compilation errors, here is my snippet:

```C++
#include <nlohmann/json.hpp>
#include <iostream>

using json = nlohmann::ordered_json;

int main(void) {
    json j = json::parse("{\"b\":2,\"a\":1}");
    std::cout<< j <<std::endl;
    return 0;
}
```

Am I using it wrong?
gcc version 8.3.1

**Reply by nlohmann** [reply]
This is a bug tracked in #2315. There is a branch linked where the bug is fixed. I will merge it to `develop` soon.
